fabfile/run.py

```diff
@@ -203,3 +202,31 @@ def tpch_queries(query_info, connectionURI, pg_version, citus_version, config_fi
         out_val = run(run_string)
         results_file.write(out_val)
         results_file.write('\n')
+
+def filter_and_put_citus_valgrind_outputs(valgrind_log_path):
+    # get stack trace id that includes calls to citus.so
+    run('cat {} | grep citus.so | awk \'{ print $1 }\' | uniq  > trace_ids'.format(valgrind_log_path))
+    # filter stack traces with stack trace ids that we found with above command
+    run('while read line; do grep {} -e $line ; done < trace_ids > {}'.format(
+        valgrind_log_path, 
+        os.path.join(config.RESULTS_DIRECTORY, config.CITUS_RELATED_VALGRIND_LOGS_FILE)))
+
+@task
+@roles('master')
+def valgrind(*args): 
+    'Runs valgrind tests'
+
+    # set citus path variable
+    repo_path = config.settings['repo_path']
+    
+    use.valgrind()
+    setup.valgrind()
+
+    with cd(os.path.join(repo_path, config.RELATIVE_REGRESS_PATH)):
+        run('make check-multi-vg VALGRIND_LOG_FILE=$VALGRIND_LOGS_FILE')
+        
+        # ship regression.diffs to result file in order to push to github
+        run('mv {} {}'.format(config.REGRESSION_DIFFS_FILE, config.RESULTS_DIRECTORY))
```

Comment:
here we should check if the file exists.

maybe:

```python
if os.path.isfile(config.REGRESSION_DIFFS_FILE):
    run('mv {} {}'.format(config.REGRESSION_DIFFS_FILE, config.RESULTS_DIRECTORY))
```

Reply:
we perform this check with the following commit https://github.com/citusdata/test-automation/pull/130/commits/381df184deddcbfd8ac41b8a1bce416beb1f6961

Reply:
maybe it would be better to do this check in pythonic way in order not to encounter `fatal error` lines in fab outputs. I will make needed the change

Reply:
done in 3b42d6ea24ca75a555ff4a0528d4f2e4c8eda409